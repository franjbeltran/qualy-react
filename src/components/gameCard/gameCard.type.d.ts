export type Game = {
    id: string,
    name: string,
    image: string,
    handleClick?: (event) => void
}