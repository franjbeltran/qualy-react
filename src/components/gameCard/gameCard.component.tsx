import { Button, Card, CardContent, CardMedia, Typography } from "@mui/material";
import { CardMediaContainer } from "./gameCard.styles";
import { Game as GameType } from "./gameCard.type";

export const GameCard = ({ id, name, image, handleClick }: GameType) => {
    return (
        <Card>
            <CardMediaContainer>
                <CardMedia data-testid='card-media' component='img' alt='' image={image} />
            </CardMediaContainer>
            <CardContent>
                <Typography variant="h4">
                    <Button data-testid='button-game' variant="text" onClick={handleClick}>
                        {name}
                    </Button>
                </Typography>
            </CardContent>
        </Card>
    );
}