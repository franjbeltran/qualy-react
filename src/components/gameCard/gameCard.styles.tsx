import styled from 'styled-components';

export const CardMediaContainer = styled.div`
    height: 200px;
    display: flex;
`
