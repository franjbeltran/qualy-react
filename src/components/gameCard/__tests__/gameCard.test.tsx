import '@testing-library/jest-dom';
import { cleanup, fireEvent, getByTestId, render, RenderResult, screen } from '@testing-library/react';
import { GameCard } from '../gameCard.component';
import { Game as GameType} from '../gameCard.type';

describe('GameCard component', () => {
    afterEach(cleanup);

    it('renders the component with all properties', () => {
        // Given
        const props: GameType = {
            id: '1',
            name: 'Test',
            image: 'test'
        };

        // When
        const { container }: RenderResult = render(<GameCard {...props} />);

        // Then
        expect(container).toMatchSnapshot();
        expect(screen.getByText(props.name)).toBeInTheDocument();
        expect(screen.getByTestId('card-media')).toBeInTheDocument();
    });

    it('calls handleGame function', () => {
        // Given
        const props: GameType = {
            id: '1',
            name: 'Test',
            image: 'test',
            handleClick: jest.fn()
        };

        // When
        render(<GameCard {...props} />);
        fireEvent.click(screen.getByTestId('button-game'));

        // Then
        expect(props.handleClick).toHaveBeenCalledTimes(1);
    });
});