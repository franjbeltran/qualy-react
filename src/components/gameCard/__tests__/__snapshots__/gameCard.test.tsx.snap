// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard component renders the component with all properties 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="sc-bcXHqe emzPsN"
    >
      <img
        alt=""
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        data-testid="card-media"
        src="test"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="button-game"
          tabindex="0"
          type="button"
        >
          Test
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </h4>
    </div>
  </div>
</div>
`;
